frontend/src/components/Profile/Profile.tsx

```diff
@@ -16,8 +16,10 @@ const { posts, followers, following } = {
 
 export function Profile() {
   const [showCreatePostModal, setShowCreatePostModal] = useState(false);
-  const id = useAppSelector(selectUserId);
-  const { data } = useGetProfileByIdQuery(id);
+  const params = useParams();
+  let username = useAppSelector(selectUserUsername);
```

Comment:
Why do we need this?

Reply:
@ejaz-hussain-1050 You marked this as resolved, but I see no response. Why do you need to read the username from the Redux store?